src/message.rs

```diff
@@ -0,0 +1,41 @@
+use crate::{Block, Key, PeerId, State};
+
+pub trait Message {
+    type Block: Block;
+    type Key: Key;
+    type PeerId: PeerId;
+
+    fn new(
+        height: u64,
+        key: Self::Key,
+        peer: Self::PeerId,
+        state: State,
+        block: Self::Block,
+    ) -> Self;
+
+    fn block(&self) -> &Self::Block;
+    fn owned_block(&self) -> Self::Block;
+    fn height(&self) -> u64;
```

Comment:
We should also have `round(&self)` in addition to height.

Reply:
It is a bit controversial because the paper won't make references to the consensus implementation of `(H, R)`, but only rounds instead. I think the best course of action is to create a trait `Round` and the implementor can use it as he wants, being that a tuple as in tendermint or a single height.

Thoughts?

Reply:
Our goal is to follow in the footsteps of Tendermint, which is more specifically suited and battle-tested for decentralized blockchain-based protocols. While pBFT is the foundation for the majority of modern approaches and can be cited, I don't think we should restrict our design to a paper from 1999.

The Tendermint paper has a clear definition of height meaning the block height, and rounds which reset to 0 at the start of each new block height.
https://tendermint.com/static/docs/tendermint.pdf

![tendermint](https://user-images.githubusercontent.com/794823/148131074-7e67e168-19f8-4c98-9215-e80cdd9b9837.jpg)

Reply:
Hmm but if we are to use the tendermint paper as base, then there is another change I need to make. According to tendermint, we propose the new block after the confirmation of `precommit`, whereas we currently do so only after `commit`. cc @adlerjohn 

Its naturally not a challenge to go for the tendermint approach, but then how about we rename the lib?

Reply:
Renaming the lib might be in order, e.g. fuelmint or something similar as Tendermint is what we had agreed on.

Reply:
To follow-up: https://github.com/FuelLabs/fuel-pbft/issues/2